rules/utils/get-number-literal-type.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+/**
+ * Return the type of the number in the node
+ * @param {Node} node - The node to check. Its type must be "Literal", and its value must be a number.
+ * @returns {undefined|"bigint"|"hexadecimal"|"binary"|"octal"|"number"}
+ */
+function getNumberLiteralType(node) {
+	if (!node?.type === 'Literal' || !['number', 'bigint'].includes(typeof node.value)) {
+		return;
+	}
+
+	if (node.bigint) {
+		return 'bigint';
+	}
```

Comment:
bigint  can  be  binary too, we  should just strip `n` and treat rest part as  number

Reply:
What do you mean?

Reply:
0b1010n

Reply:
Yes but what do you mean by strip the `n` and treat the rest as a number?
Do you mean that if there is already a prefix, such as `0b` then we just completely remove the `n`?

Reply:
I mean you should just strip `n` and return the "type" as `binary`

We don't need option for `bigint`, `0b1010n` should follow `binary` options.

Reply:
> strip n 

I mean, like this

https://github.com/sindresorhus/eslint-plugin-unicorn/pull/834/files#diff-d6dd6615153d8c4ed594a894e169f055R22

Reply:
Oh i just finished that and I did not do it like this... Check my last commit (will be pushed in a couple of minutes) and leave a review if you want me to do it like this 😄